syntax = "proto3";

option go_package = "proto_gen/identityservicepb";

service UserAuth {
    rpc Register (UserInfo) returns (Confirmation);
    rpc Login (UserCredentials) returns (LoginConfirmation);
    rpc Balance (UserLogin) returns (Amount);
    rpc Deposit (Transaction) returns (Confirmation);
    rpc Withdraw (Transaction) returns (Confirmation);
}

message UserInfo {
    string username = 1;
    string password = 2;
}

message UserCredentials {
    string username = 1;
    string password = 2;
}

message UserLogin {
    string login = 1;
}

message LoginConfirmation {
  bool success = 1;
  string token = 2;
  string message = 3;
}

message Amount {
    double value = 1;
}

message Transaction {
    string login = 1;
    int32 amount = 2;
}

message Confirmation {
    bool success = 1;
    string message = 2;
}
